server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocatorTests.java

```diff
@@ -167,23 +154,20 @@ public void afterPrimariesBeforeReplicas(RoutingAllocation allocation) {}
         final var indexMetadata = createIndex(TEST_INDEX);
         var clusterState = createClusterState(discoveryNode, indexMetadata);
 
-        var listener = new TestActionListener();
```

Comment:
I would prefer to keep the assertions that this listener is called (eventually) although I would not be surprised if we need to move those assertions later.